### DISCUSSION TITLE
Submitting New Inventory for Existing Products via UPC

### DISCUSSION THREAD

**Discussion by jayhardacre** [abuzuhri/Amazon-SP-API-CSharp#174] [Q&A]
This isnt an issue, more of an inquiring. The only thing I have left in my system to do with your api is send up new products to sell. At the moment we send up a "POST_FLAT_FILE_INVLOADER_DATA" file (string) generated in memory and get a FeedSubmissionID that we check later in the night to see how successful we are. These are all linked via UPC (or at least try) in amazon's catalog.

I cannot see that anywhere in the feeds sample code. Would you be able to have a sample code of this?

Anything would help.
Thanks

**Comment by abuzuhri** [comment]
Do you mean create new product from scratch or just map with product already exist before in Amazon ?

**Reply by jayhardacre** [reply]
Map products that already exist in Amazon catalog.

**Reply by abuzuhri** [reply]
It's simple but many steps need to call 


This for MWS but it's same steps you need to do , check it's if step not work with you I will send full code when I'm in the front of my laptop 

https://docs.developer.amazonservices.com/en_UK/fba_guide/FBAGuide_CreateListing.html

**Reply by jayhardacre** [reply]
We don't do FBA. Just a little background, we use VB, and call with MWS dll. Here is the current code:

![image](https://user-images.githubusercontent.com/104643115/166318738-d770435d-aeed-4a0b-9427-54cdb3c16d89.png)

This is what I have so far with this SDK, I dont know if this is right, and there is no place for the seller id (if it is needed):

![image](https://user-images.githubusercontent.com/104643115/166322250-756f56c7-a658-4fd8-905a-9b976c5995fe.png)

Is this the way I should do it?

**Reply by abuzuhri** [reply]
> We don't do FBA. Just a little background, we use VB, and call with MWS dll. Here is the current code:
> 
> ![image](https://user-images.githubusercontent.com/104643115/166318738-d770435d-aeed-4a0b-9427-54cdb3c16d89.png)
> 
> This is what I have so far with this SDK, I dont know if this is right, and there is no place for the seller id (if it is needed):
> 
> ![image](https://user-images.githubusercontent.com/104643115/166322250-756f56c7-a658-4fd8-905a-9b976c5995fe.png)
> 
> Is this the way I should do it?



I only speaking about the step you need to do mapping existing product 

1- Submit the Product Feed (POST_PRODUCT_DATA)
2- Submit the Product Feed (POST_PRODUCT_PRICING_DATA)
3- Submit the Product Feed (POST_INVENTORY_AVAILABILITY_DATA)

**Reply by jayhardacre** [reply]
Are these the steps to post new product data to our inventory that already exists in Amazon's catalog?

**Reply by abuzuhri** [reply]
Yes , those feeds to do mapping for existing products
1- POST_PRODUCT_DATA
2- POST_INVENTORY_AVAILABILITY_DATA
3- POST_PRODUCT_PRICING_DATA

just try to do those and if its not work i will help you 

* adding existing product in Amazon name it *mapping*

**Reply by jayhardacre** [reply]
So I got it to work via uploading the flat file. This is my method:

![image](https://user-images.githubusercontent.com/104643115/166532020-d32432df-0ecc-4e92-ade0-e3060228b76d.png)

Build the flat file, in this case it's a tab delimited file.

![image](https://user-images.githubusercontent.com/104643115/166533504-868a074c-a359-4c4e-972f-bcaa49cc23b9.png)

And the HttpPutRequest looks like this:

![image](https://user-images.githubusercontent.com/104643115/166535057-8ca84e5c-9510-461d-80c0-8b3b4d443193.png)

**Reply by abuzuhri** [reply]
I'm not sour which library you use to generate XML or how to submit feed not expert in VB
